keps/sig-auth/20190607-certificates-api.md

```diff
@@ -122,19 +128,39 @@ This is typical of many PKI architectures.
 
 ### Sequence of an Issuance
 
-A typical successful issuance proceeds as follows.
+A typical successful issuance proceeds as follows:
 
 ![CSR](/keps/sig-auth/csr.png)
 
 1. The requestor generates a private key, builds a certificate signing request,
    and submits the `CertificateSigningRequest` to the Kubernetes certificates
    API.
 1. The approver controller observes the newly submitted request, validates and
-   authorizes the request and if all goes well, approves the request.
+   authorizes the request and adds an `Approved` condition via the `/approval` subresource.
 1. The signer observes the approval, mints a new certificate and stores it in
-   the `.Status.Certificate` field.
+   the `.Status.Certificate` field via the `/status` subresource.
 1. The requestor observes the update, and stores the certificate locally.
 
+Approver failure sequence:
+
+1. The requestor generates a private key, builds a certificate signing request,
+   and submits the `CertificateSigningRequest` to the Kubernetes certificates
+   API.
+1. The approver controller observes the newly submitted request, determines the
+   request should be denied, and adds a `Denied` condition via the `/approval` subresource.
+1. The requestor observes the `Denied` condition.
+
+Signer failure sequence:
+
+1. The requestor generates a private key, builds a certificate signing request,
+   and submits the `CertificateSigningRequest` to the Kubernetes certificates
+   API.
+1. The approver controller observes the newly submitted request, validates and
+   authorizes the request and adds an `Approved` condition via the `/approval` subresource.
+1. The signer observes the approval, but encounters a terminal failure during the signing process,
```

Comment:
it's not obvious to me why this isn't denied==true.  Is this pre-existing?

Reply:
A signing failure is not the same as a denial. We don't currently have a way for a signer to indicate permanent failure signing an approved CSR.

Reply:
> A signing failure is not the same as a denial. We don't currently have a way for a signer to indicate permanent failure signing an approved CSR.

Yeah, got my actors messed up